src/multivariate/dirichlet.jl

```diff
@@ -375,3 +375,63 @@ function fit_mle(::Type{<:Dirichlet}, P::AbstractMatrix{Float64},
     elogp = mean_logp(suffstats(Dirichlet, P, w))
     fit_dirichlet!(elogp, α; maxiter=maxiter, tol=tol, debug=debug)
 end
+
+## Differentiation
+function ChainRulesCore.frule((_, Δalpha)::Tuple{Any,Any}, DT::Union{Type{Dirichlet{T}}, Type{Dirichlet}}, alpha::AbstractVector{T}; check_args = true) where {T}
+    d = DT(alpha; check_args=check_args)
+    Δalpha = ChainRulesCore.unthunk(Δalpha)
+    ∂alpha0 = sum(Δalpha)
+    ∂lmnB::typeof(∂alpha0) = sum(Δalpha[i] * (SpecialFunctions.digamma(alpha[i]) - SpecialFunctions.digamma(d.alpha0)) for i in eachindex(alpha))
+    backing = (alpha=Δalpha, alpha0=∂alpha0, lmnB=∂lmnB)
+    t = ChainRulesCore.Tangent{typeof(d), NamedTuple{(:alpha, :alpha0, :lmnB), Tuple{typeof(alpha), typeof(d.alpha0), typeof(d.lmnB)}}}(backing)
+    return d, t
```

Comment:
This can be simplified to
```suggestion
    tangent = ChainRulesCore.Tangent{typeof(d)}(; alpha=Δalpha, alpha0=∂alpha0, lmnB=∂lmnB)
    return d, tangent
```

Reply:
this currently fails inference for some reason :/ hence the weird rewrite

Reply:
Weird, this should be exactly the same: https://github.com/JuliaDiff/ChainRulesCore.jl/blob/2d75b4be102bb41ba3ac6df6dec8bb9617b20f0f/src/tangent_types/tangent.jl#L42-L45

Reply:
Does type inference fail only on some Julia versions or is it version independent?

Reply:
it failed at least on 1.8

Reply:
I just checked locally and there are no type inference issues with the suggestion above. I'll run the full tests as well.

Reply:
I could confirm my impression, tests pass with older and recent Julia versions: https://github.com/JuliaStats/Distributions.jl/pull/1555